app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :logged_in?, only: %i[new create]
```

Comment:
I noticed you do have a before action, but when are you using it? Does it accomplish what you intend?

Maybe a method that redirects may be more useful.